src/core/socialprovider.js

```diff
@@ -182,16 +182,21 @@ XMPPSocialProvider.prototype.connect = function(continuation) {
     // This may indicate a broken connection to XMPP.
     // TODO: handle this.
     this.logger.error('received unhandled close event', e);
+    if (this.status === 'ONLINE') {
+      this.ping_();
```

Comment:
What does this do? add comment & remove TODO if this handles connection breaking.